modules/highgui/src/write_mat_to_xrgb8888.simd.hpp

```diff
@@ -0,0 +1,129 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include "opencv2/core/hal/intrin.hpp"
+
+
+namespace cv {
+CV_CPU_OPTIMIZATION_NAMESPACE_BEGIN
+
+void write_mat_to_xrgb8888(cv::Mat const &img, void *data);
+
+#ifndef CV_CPU_OPTIMIZATION_DECLARATIONS_ONLY
+
+void write_mat_to_xrgb8888(cv::Mat const &img, void *data) {
+    CV_CheckTrue(data != nullptr, "data must not be nullptr.");
+    CV_CheckType(img.type(), (img.type() == CV_8UC1) || (img.type() == CV_8UC3) || (img.type() == CV_8UC4),
```

Comment:
Checks should be done outside of optimized functions.

Reply:
Thank you for your comment, I removed redandancy checks.